legacy/messaging/src/main/java/org/jboss/as/messaging/MigrateOperation.java

```diff
@@ -494,12 +495,16 @@ private boolean connectionFactoryIsUsingInVMConnectors(OperationContext context,
     private void migrateDiscoveryGroup(ModelNode newAddOp, List<String> warnings) {
         // These attributes are not present in the messaging-activemq subsystem.
         // Instead a socket-binding must be used to configure the broadcast-group.
+        boolean mandatorySocketBinding = false;
         for (String property : asList(LOCAL_BIND_ADDRESS.getName(), GROUP_ADDRESS.getName(), GROUP_PORT.getName())) {
             if (newAddOp.has(property)) {
                 newAddOp.remove(property);
                 warnings.add(ROOT_LOGGER.couldNotMigrateDiscoveryGroupAttribute(property, pathAddress(newAddOp.get(OP_ADDR))));
+                mandatorySocketBinding = true;
             }
         }
+        if (mandatorySocketBinding)
+            newAddOp.get(SOCKET_BINDING.getName()).set("group-x-binding");
```

Comment:
But how the migrate operation determines new socket-binding attributes (name ,interface, port) for user?
It will be a warning to use a real socket-binding instead of unsupported attributes, and user should update the configuration as per migration warnings as their needs.
group-x-binding only satisfies the validation since socket-binding and jgroup-channel are alternatives attributes after change in CommonAttributes.java. Same as below for jgroup-channe, one of jgroup-channel and socket-binding needs to present in case of the alternative is missing.